improving-permSpace

tesonep opened the pull request:
- In bulk move to perm space we generate the remembered set at the end only
- Adding Statistics
- Adding tests

tesonep committed:
- Adding tests
- In bulk move to perm space we generate the remembered set at the end only
- Adding Statistics

tesonep committed:
Fixing Windows build

guillep commented on smalltalksrc/VMMaker/VMRememberedSet.class.st:
```diff
@@ -137,7 +152,7 @@ VMRememberedSet >> growRememberedSet [
 		[newObj := manager allocatePinnedSlots: numSlots + 1024.
 		 newObj ifNil:
-			[(manager growOldSpaceByAtLeast: numSlots + 1024) ifNil: [self error: 'could not grow remembered set'].
+			[(manager growOldSpaceByAtLeast: (numSlots + 1024) * self wordSize) ifNil: [self error: 'could not grow remembered set'].
```
Nice catch here

guillep reviewed (approved):
I did a superficial review, it looks ok, I don't have enough knowledge to make a deeper one. I trust your tests ;)

tesonep committed:
Adding additional Remembered set from Perm Space to New Space

tesonep committed:
Fixing tests and adding new remembered set

tesonep committed:
- Adding duplicates when migrating form Perm->New RememberedSet to Perm->Old
- Removing duplicates using the mark bit
- Adding statistics for the Perm->New Remembered Set

tesonep committed:
- Invalidating Become of PermanentObjects.
- Improving the cleanup of Perm to old remembered set.
- Handling duplicates in the Perm to old remembered set.
